src/Features/Core/Portable/SolutionCrawler/WorkCoordinator.cs

```diff
@@ -184,6 +205,28 @@ public void Reanalyze(IIncrementalAnalyzer analyzer, ReanalyzeScope scope, bool
                 }
             }
 
+            private void OnActiveDocumentChanged(object sender, DocumentId activeDocumentId)
+            {
+                IAsyncToken asyncToken;
+                if (SolutionCrawlerOptions.GetBackgroundAnalysisScope(_registration.Workspace.Options) == BackgroundAnalysisScope.ActiveFile &&
+                    activeDocumentId != null)
```

Comment:
wait.  if activeDocumentId became null, why are we ignorign that?

Reply:
We are just treating it as a no-op for solution crawler when user moves from a source document to a non-source document. As soon as user switches to a source document, all the correct analysis callbacks will happen.

Reply:
1. woudl love to doc that.
2. would love even more to have tests that validate that that works properly :)